pandas/core/indexes/base.py

```diff
@@ -649,11 +649,11 @@ def _engine(self):
     # Array-Like Methods
 
     # ndarray compat
-    def __len__(self):
+    def __len__(self) -> int:
         """
         Return the length of the Index.
         """
-        return len(self._data)
+        return len(self._data)  # type: ignore
```

Comment:
if adding `# type: ignore` can you add the mypy github issue number and mypy error message as comments.

Reply:
sure.  the problem is that _data is None at the class level, and it wants Sized.   maybe once py35 is dropped we can annotate `_data: Sized`?

Reply:
ah. OK. I think the idiomatic way to deal with this scenario from a mypy perspective is to precede the return statement with an `assert self._data is not None`. see https://mypy.readthedocs.io/en/latest/kinds_of_types.html#optional-types-and-the-none-type

and/or add a TODO to remove

Reply:
updated

Reply:
3.5 drop was merged !

Reply:
excellent.  that shouldnt affect this PR much, but ill take a look at the core.apply one to see if it can be simplified